### ISSUE TITLE
[BERT/PyTorch] BERT PyTorch performance conflict in README? Looking for explanation for 30% throughput degradation.

### ISSUE THREAD

**Issue by manualcomplete** [NVIDIA/DeepLearningExamples#939]
**BERT PyTorch Documentation throughput mismatch**

While reviewing the results of Training performance for NVIDIA DGX-1 (8x V100 16G), I found what appears to be a discrepancy between the throughput results shown in this [table1](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/LanguageModeling/BERT#training-performance-nvidia-dgx-1-8x-v100-16g) and this [table2](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/LanguageModeling/BERT#pre-training-on-multiple-nvidia-dgx-1-with-16g)

The first table shows (removed irrelevant lines):
 
| GPUs | Batch size / GPU (FP32 and FP16) | Accumulation steps (FP32 and FP16) | Sequence length | Throughput - FP32(sequences/sec) | Throughput - mixed precision(sequences/sec) | Throughput speedup (FP32 - mixed precision) | Weak scaling - FP32 | Weak scaling - mixed precision
|------------------|----------------------|----------------------|-------------------|-----------------------------------------------|------------------------------------|---------------------------------|----------------------|----------------------------------------------
|8 | 8192 and 8192  | 1024 and 512| 128| 313 |1236 | 3.94| 7.83| 7.54
| 8| 4096 and 4096 | 2048 and 1024| 512| 71 |263 | 3.70| 7.89| 7.74

The second table shows (removed irrelevant lines):

| Nodes | GPUs | Batch size / GPU (FP32) | Batch size / GPU (FP16) | Sequence length | Throughput - FP32(sequences/sec) | Throughput - mixed precision(sequences/sec) | Throughput speedup (FP32 - mixed precision) | Weak scaling - FP32 | Weak scaling - mixed precision
|------------------|----------------------|----------------------|-------------------|-----------------------------------------------|------------------------------------|---------------------------------|----------------------|----------------------------------------------|--------------
|1 |8 | N/A | 16| 128| N/A |874.24 |N/A |N/A | 1.00
|1 |8 | N/A | 4| 512| N/A |195.93 |N/A |N/A | 1.00

Both tables show training performance of BERT-Large on DGX-1 16GB, the difference being the first table showed scaling of training performance with respect to number of GPUs and second table shows scaling with respect to number of nodes. As shown, there is a very large gap in terms of mixed precision throughput: **1236 seq/s** vs **874.24 seq/s**, as well as **263 seq/s** vs **195.93 seq/s**.

I would like to understand the reason for the ~30% degradation in the second set of experiments. The notable differences provided by the tables are that they use different Nvidia docker container versions (20.06-py3 vs 19.07-py3), they also use different global batch sizes, although the mini batch sizes are the same between the runs (16, 4).

Any insights on this would be appreciated!

**Comment by sharathts** [comment]
The first table includes gradient accumulation, which is used during training and shows actual training throughput. 

The second table shows raw throughout setting gradient accumulation steps to 1. More the number of accumulation steps, more the performance since weight updates are skipped.

**Comment by manualcomplete** [comment]
I see, thank you.
